elements_imaging/imaging.py

```diff
@@ -152,71 +151,81 @@ class Processing(dj.Computed):
+        self.insert1(key)
 
-            elif method == 'caiman':
-                from .readers import caiman_loader
 
-                data_dir = pathlib.Path(get_caiman_dir(key))
-                loaded_caiman = caiman_loader.CaImAn(data_dir)
+@schema
+class Curation(dj.Manual):
+    definition = """
+    -> Processing
+    curation_id: int
+    ---
+    curation_time: datetime             # time of generation of this set of curated results 
+    curation_output_dir: varchar(255)   # output directory of the curated results, relative to root data directory
+    manual_curation: bool               # has manual curation been performed on this result?
+    curation_note='': varchar(2000)  
+    """
 
-                key = {**key, 'proc_completion_time': loaded_caiman.creation_time,
-                              'proc_curation_time': loaded_caiman.curation_time}
+    def create1_from_processing_task(self, key, is_curated=False, curation_note=''):
+        """
+        A convenient function to create a new corresponding "Curation" for a particular "ClusteringTask"
+        """
+        if key not in Processing():
+            raise ValueError(f'No corresponding entry in Clustering available for: {key}; do `Processing.populate(key)`')
 
-                # Insert file(s)
-                root = pathlib.Path(scan.get_imaging_root_data_dir())
-                output_files = [loaded_caiman.caiman_fp.relative_to(root).as_posix()]
-            else:
-                raise NotImplementedError('Unknown method: {}'.format(method))
+        method, loaded_result = get_loader_result(key, ProcessingTask)
 
-            self.insert1(key)
-            self.ProcessingOutputFile.insert([{**key, 'file_path': f} for f in output_files], ignore_extra_fields=True)
+        if method == 'suite2p':
+            loaded_suite2p = loaded_result
+            curation_time = loaded_suite2p.creation_time
+        elif method == 'caiman':
+            loaded_caiman = loaded_result
+            curation_time = loaded_caiman.creation_time
+        else:
+            raise NotImplementedError('Unknown method: {}'.format(method))
 
-        elif task_mode == 'trigger':
-            start_time = datetime.now()
-            # trigger Suite2p or CaImAn here
-            # wait for completion, then insert with "completion_time", "start_time", no "curation_time"
-            return
+        # Synthesize curation_id
+        curation_id = dj.U().aggr(self & key, n='ifnull(max(curation_id)+1,1)').fetch1('n')
+        self.insert1({**key, 'curation_id': curation_id,
+                      'curation_time': curation_time, 'curation_output_dir': output_dir,
+                      'manual_curation': is_curated,
+                      'curation_note': curation_note})
 
 
 # -------------- Motion Correction --------------
 
 @schema
 class MotionCorrection(dj.Imported):
     definition = """ 
-    -> Processing
+    -> Curation
```

Comment:
Does MotionCorrection result also get curated? I thought only the segmentation results are curated.